changelogs/unreleased/4332-auto-recompile-wait-env-setting.yml

```
@@ -0,0 +1,7 @@
+description: Add environment setting to set the auto-recompile-wait time instead of the config option
+issue-nr: 4332
+change-type: patch
+destination-branches: [master]
+sections:
+  feature: "{{description}}",
+  deprecation-note: "The auto-recompile-wait option in the server configuration is now deprecated in favor of the environment setting with the same name"
```

I don't think it makes much sense to have both a `feature` and `deprecation-note` entry.
```suggestion
  deprecation-note: "The auto-recompile-wait option in the server configuration is now deprecated in favor of the environment setting with the same name"
```